app/src/main/java/com/example/sharingang/UserProfileFragment.kt

```diff
@@ -178,6 +183,31 @@ class UserProfileFragment : Fragment() {
         }
     }
 
+    private fun setupReportButton() {
+        var username = ""
+        if(currentUserId != null) {
+            binding.btnReport.visibility = View.VISIBLE
+            lifecycleScope.launch(Dispatchers.IO) {
+                username = userRepository.get(shownUserProfileId!!)!!.name
+            }
+        }
+
+        binding.btnReport.setOnClickListener {
+            lifecycleScope.launch(Dispatchers.IO) {
+                val reporterUser = userRepository.get(currentUserId!!)!!
+                val reportedUser = userRepository.get(shownUserProfileId!!)!!
+                userRepository.report(reportedUser, reporterUser)
+            }
```

Comment:
Why don't you create the function in the userViewModel rather than the user repository?

Reply:
So you prefer something like this:
```
fun reportUser(reportedUser: User, reporterUser: User)  {
    lifecycleScope.launch(Dispatchers.IO) {
            val reporterUser = userRepository.get(currentUserId!!)!!
            val reportedUser = userRepository.get(shownUserProfileId!!)!!
            userRepository.report(reportedUser, reporterUser)
    }
}
```
inside UserViewModel and then call `viewModel.report(reportedUser, reporterUser)` from the UserProfileFragment?

Reply:
Yes that's essentially what I mean